src/appleseed/renderer/kernel/lighting/materialsamplers.cpp

```diff
@@ -131,14 +131,15 @@ const ShadingPoint& BSDFSampler::trace(
         m_shading_point.get_ray().m_depth + 1);
     ray.copy_media_from(m_shading_point.get_ray());
 
-    const ShadingPoint& shading_point =
-        shading_context.get_tracer().trace<const ShadingPoint&>(
-            shading_context,
-            m_shading_point,
-            ray,
-            transmission);
-    if (shading_point.hit()) transmission.set(0.0f);
-    return shading_point;
+    const ShadingPoint* shading_point;
+    shading_context.get_tracer().trace(
```

Comment:
Yes, I also like that version more. But we need to change the name of this function (because they differ only in return values). `trace_full` and `trace__between_full`?